### ISSUE TITLE
What is the difference between PatchTSMixerForPretraining and PatchTSMixerForPrediction classes?

### ISSUE THREAD

**Issue by gabriel301** [ibm-granite/granite-tsfm#300]
Reviewing the doc in https://huggingface.co/docs/transformers/main/en/model_doc/patchtsmixer I found this PatchTSMixerForPretraining class and the "only" difference I could see is that it is used for "mask pretraining".

Could anyone explain what is the mask pretraining, that the differences between PatchTSMixerForPretraining and PatchTSMixerForPrediction?

Can I use PatchTSMixerForPretraining  for training a model and the forecast values?

Thanks!

**Comment by wgifford** [comment]
Hi @gabriel301 thanks for your interest in TSFM.

In general, we have found that pretraining with a forecasting objective (rather than masked pretraining) yielded better results when the downstream task is forecasting. We have a sequence of notebooks for the PatchTST model that shows how masked pretraining works here: https://github.com/ibm-granite/granite-tsfm/tree/main/notebooks/thinklab
Essentially the process is 1) pretrain using masked pretraining, 2) load the pretrained model into a model ForPrediction, and 3) finetune using a forecasting objective. You could try a similar process for TSMixer.

**Comment by gabriel301** [comment]
Thanks @wgifford ! I review the notebook.
